src/globus_sdk/transport/retry.py

```diff
@@ -72,8 +72,8 @@ def set_retry_check_flags(flag: RetryCheckFlags):
     >>> def foo(ctx): ...
     """
 
-    def decorator(func):
-        func._retry_check_flags = flag
+    def decorator(func: Callable) -> Callable:
+        func._retry_check_flags = flag  # type: ignore
```

Comment:
This reminds me. If we're pulling in `typing_extensions`, we can use `Protocol`. That lets us do an alternate fix in these cases (function attribute addition) in which we cast the func to a protocol class. Not sure if it's worth it.

But also, I'm 👍 on using an ignore for this -- I think I've done so elsewhere for the same purpose.